Check linting in branches (not main) (#73)
NMDSdevopsServiceAdm/DataEngineering: README.md

Once the workflow has completed AWS will contain all the infrastructure required to run the pipeline and all associated glue jobs.<br>

Environments must be torn down manually once a git branch has been deleted. This is a simple process, [detailed below](#destroying-terraform).

> ❗ **When merging with the main branch**: The workflow will run here too. There is a mandatory, manual approval step required here. Please read the output of `terraform plan`. Ensure this is correct, then give your approval. The workflow will complete and the main (production) infrastructure will be updated. The main branch workflows can be found [here](https://app.circleci.com/pipelines/github/NMDSdevopsServiceAdm/DataEngineering?branch=main&filter=all).

<br>

## Continuous Integration
***The CI part of [CICD](https://www.redhat.com/en/topics/devops/what-is-ci-cd#:~:text=CI%2FCD%20is%20a%20method,continuous%20delivery%2C%20and%20continuous%20deployment.)***

When you push to a remote git branch, we run some linting checks for Python and terraform code as a part of the CircleCi workflow (mentioned above).

> If your branch fails either of these checks it means you need to run the relevent linter to fix the errors.

Reviewer: ```suggestion
If your branch fails either of these checks, you need to run the relevant linter to fix the errors.
```

> Instructions for both terraform and python are detailed below.

Reviewer: ```suggestion
Instructions for both Terraform and Python are detailed below.
```

> ### Linting python code

Reviewer: ```suggestion
### Linting Python code
```

We use [black](https://black.readthedocs.io/en/stable/) to lint our python code.
Install black using pip
```
pip install black
```

> To lint all the python files, first ensure you're at the root of the repository then run

Reviewer: ```suggestion
To lint all the Python files, first ensure you're at the root of the repository, then run
```

```
black .
```

> ### Linting terraform code

Reviewer: ```suggestion
### Linting Terraform code
```

Install terraform following [the instructions below](#installing-terraform).

> Ensure you are at the root of the repository then run

Reviewer: ```suggestion
Ensure you are at the root of the repository, then run
```
